packages/utils/telemetry-utils/src/mockLogger.ts

```diff
@@ -259,3 +264,32 @@ function matchObjects(actual: ITelemetryPropertiesExt, expected: ITelemetryPrope
 	}
 	return true;
 }
+
+/**
```

Comment:
For reviewers: if we are happy with the overall testing pattern introduced in this PR, I'll split these changes out into a separate PR.

Reply:
LGTM but you'll likely want to get review from someone else for this part.

Reply:
Extracted into #21466